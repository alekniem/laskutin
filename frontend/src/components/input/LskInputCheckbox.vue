<template>
    <ValidationProvider
        :customMessages="{ required: trans('Required') }"
        :rules="{ required: { allowFalse: false } }"
        :vid="vid"
        ref="provider"
        tag="div"
        #default="{ errors, valid }"
    >
        <v-checkbox
            @change="$emit('change', $event)"
            :error-messages="errors"
            :id="cid"
            :input-value="checked"
            :label="trans(label)"
            :success="valid"
        />
    </ValidationProvider>
</template>

<script>
import { ValidationProvider } from "vee-validate";

export default {
    name: "LskInputCheckbox",

    components: {
        ValidationProvider,
    },

    model: {
        prop: "checked",
        event: "change",
    },

    props: {
        cid: { type: String, required: true },
        checked: { type: Boolean, required: true },
        label: { type: String, required: true },
        vid: { type: String, required: true },
    },
};
</script>
