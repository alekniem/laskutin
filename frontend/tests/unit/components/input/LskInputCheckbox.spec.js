import flushPromises from "flush-promises";
import LskInputCheckbox from "@/components/input/LskInputCheckbox";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskInputCheckbox", () => {
    it("should match snapshot", async () => {
        const wrapper = createWrapper(LskInputCheckbox, {
            propsData: {
                cid: "TEST_CID",
                checked: true,
                label: "TEST LABEL",
                vid: "TEST_VID",
            },
        });
        await flushPromises();
        expect(wrapper.html()).toMatchSnapshot("valid input");
    });

    it("should show an error message", async () => {
        const wrapper = createWrapper(LskInputCheckbox, {
            propsData: {
                cid: "TEST_CID",
                checked: false,
                label: "TEST LABEL",
                vid: "TEST_VID",
            },
        });
        await flushPromises();
        wrapper.vm.$refs.provider.validate();
        await flushPromises();
        expect(wrapper.find(".v-messages__message").text()).toBe("Required");
    });
});
