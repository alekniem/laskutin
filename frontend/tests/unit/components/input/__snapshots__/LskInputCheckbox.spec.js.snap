// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskInputCheckbox should match snapshot: valid input 1`] = `
<div>
  <div class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--checkbox success--text">
    <div class="v-input__control">
      <div class="v-input__slot">
        <div class="v-input--selection-controls__input">
          <i aria-hidden="true"
             class="v-icon notranslate mdi mdi-checkbox-marked theme--light success--text"
          >
          </i>
          <input aria-checked="true"
                 id="TEST_CID"
                 role="checkbox"
                 type="checkbox"
                 value
          >
          <div class="v-input--selection-controls__ripple success--text">
          </div>
        </div>
        <label for="TEST_CID"
               class="v-label theme--light success--text"
               style="left: 0px; position: relative;"
        >
          TEST LABEL
        </label>
      </div>
      <div class="v-messages theme--light success--text">
        <transition-group-stub tag="div"
                               name="message-transition"
                               class="v-messages__wrapper"
        >
        </transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
